### PR TITLE
Fix RPN loss normalization and topk parameter

### PR DESCRIPTION
**Pull request by karakusc**
Converges under bs=2 duplicate image case with

[0416 10:16:11 @monitor.py:467] mAP(bbox)/IoU=0.5: 0.5938
[0416 10:16:11 @monitor.py:467] mAP(bbox)/IoU=0.5:0.95: 0.37597
[0416 10:16:11 @monitor.py:467] mAP(bbox)/IoU=0.75: 0.40791
[0416 10:16:11 @monitor.py:467] mAP(bbox)/large: 0.48862
[0416 10:16:11 @monitor.py:467] mAP(bbox)/medium: 0.41575
[0416 10:16:11 @monitor.py:467] mAP(bbox)/small: 0.21882
[0416 10:16:11 @monitor.py:467] mAP(segm)/IoU=0.5: 0.56195
[0416 10:16:11 @monitor.py:467] mAP(segm)/IoU=0.5:0.95: 0.34604
[0416 10:16:11 @monitor.py:467] mAP(segm)/IoU=0.75: 0.36479
[0416 10:16:11 @monitor.py:467] mAP(segm)/large: 0.47089
[0416 10:16:11 @monitor.py:467] mAP(segm)/medium: 0.38222
[0416 10:16:11 @monitor.py:467] mAP(segm)/small: 0.185

Under normal bs=2 with padding, segm converges but bbox accuracy is slightly off:
[0416 08:50:55 @monitor.py:467] mAP(bbox)/IoU=0.5: 0.58983
[0416 08:50:55 @monitor.py:467] mAP(bbox)/IoU=0.5:0.95: 0.37095
[0416 08:50:55 @monitor.py:467] mAP(bbox)/IoU=0.75: 0.40082
[0416 08:50:55 @monitor.py:467] mAP(bbox)/large: 0.48156
[0416 08:50:55 @monitor.py:467] mAP(bbox)/medium: 0.40704
[0416 08:50:55 @monitor.py:467] mAP(bbox)/small: 0.21482
[0416 08:50:55 @monitor.py:467] mAP(segm)/IoU=0.5: 0.55738
[0416 08:50:55 @monitor.py:467] mAP(segm)/IoU=0.5:0.95: 0.34289
[0416 08:50:55 @monitor.py:467] mAP(segm)/IoU=0.75: 0.36383
[0416 08:50:55 @monitor.py:467] mAP(segm)/large: 0.46323
[0416 08:50:55 @monitor.py:467] mAP(segm)/medium: 0.37665
[0416 08:50:55 @monitor.py:467] mAP(segm)/small: 0.18245

### CODE REVIEW

File: MaskRCNN/model_fpn.py [hunk 1]
```diff
@@ -285,7 +285,7 @@ def generate_fpn_proposals_batch_tf_op(multilevel_anchor_boxes,
     all_boxes = []
     all_scores = []
     if cfg.FPN.PROPOSAL_MODE == 'Level':
-        fpn_nms_topk = cfg.RPN.TRAIN_PER_LEVEL_NMS_TOPK if training else cfg.RPN.TEST_PER_LEVEL_NMS_TOPK
+        fpn_nms_topk = cfg.RPN.TRAIN_PER_LEVEL_NMS_TOPK*batch_size if training else cfg.RPN.TEST_PER_LEVEL_NMS_TOPK
```
Comment (armandmcqueen) [inline MaskRCNN/model_fpn.py:288]: Are you sure this is right? You pick top-k per image, not top-k per batch?

Comment (karakusc) [reply MaskRCNN/model_fpn.py:288]: The `rois` that we return here are pooled across the images in the batch, so I think it makes sense to scale the number of ROIs with batch size. When I did the tests with duplicate image, the fastrcnn box loss is properly normalized only when I do this scaling.

Comment (karakusc) [reply MaskRCNN/model_fpn.py:288]: i.e., this line: https://github.com/armandmcqueen/tensorpack-mask-rcnn/blob/432ad8072efedb880f4a66260b4c81919e7fd5e2/MaskRCNN/model_fpn.py#L370

Comment (armandmcqueen) [reply MaskRCNN/model_fpn.py:288]: Okay - that's different than what PyTorch does, but they have a TODO note about changing the code to do top-k per image so I think it's fine.

Comment (armandmcqueen) [reply MaskRCNN/model_fpn.py:288]: Okay - that's different than what PyTorch does, but they have a TODO note about changing the code to do top-k per image so I think it's fine.

**Comment by armandmcqueen** [comment]
Looks good. Approved
